Makefile

```diff
@@ -80,33 +81,35 @@ builtin_fc_print_helpers builint_fc_find_index builtin_fc_substitute \
 builtin_fc_edit \
 signal_handle_child_death
 TESTS = unit_test builtin_assign_test
-OBJECTS := $(SRCS:%=%.o)
+OBJECTS := $(SRCS:%=$(OBJDIR)%.o)
 TESTOBJECTS := $(TESTS:%=%.o)
 SRCS := $(SRCS:%=%.c)
 TESTS := $(TESTS:%=%.c)
 
-all: $(OBJECTS) $(LIBFT) $(NAME)
+all: $(LIBFT) $(NAME)
 
-$(NAME): $(OBJECTS) main.o
-	@$(CC) $(FLAGS) $^ $(COVERAGE) $(INCLUDES) $(LIB) -o $(NAME)
+$(NAME): $(OBJDIR) $(OBJECTS) $(OBJDIR)main.o
+	@$(CC) $(FLAGS) $(OBJECTS) $(OBJDIR)main.o $(COVERAGE) $(INCLUDES) $(LIB) -o $(NAME)
 	@echo "[ + ] vsh has been compiled"
 
-%.o: %.c vsh.h
+$(OBJDIR)%.o: %.c vsh.h
 	@$(CC) -o $@ $(FLAGS) $< $(COVERAGE) $(INCLUDES) -c
 
 $(LIBFT):
 	@$(MAKE) -C libft
 
+$(OBJDIR):
+	@mkdir $(OBJDIR)
+
 clean:
-	@rm -f $(OBJECTS) $(TESTOBJECTS) main.o
-	@$(MAKE) -C libft clean
+	@rm -f $(OBJECTS) $(TESTOBJECTS) $(OBJDIR)main.o
+	@$(MAKE) -C libft fclean
```

Comment:
Why this change?

Reply:
To paraphrase @marijnvanh: "We want to clean all dependencies of vsh and libft is one of those"